mallow_notifications/sns/notification.py

```diff
@@ -0,0 +1,171 @@
+"""This module provides the `Notification` class, which is used to send
+notifications using the Amazon SNS service."""
+
+from mallow_notifications.base.exceptions import NotificationError
+from mallow_notifications.base.logger import get_logger
+from mallow_notifications.sns import (
+    SNSPublishMessage,
+    SNSPushNotification,
+    SNSSandboxSMS,
+    SNSTopics,
+    SNSTopicSubscribe,
+)
+
+logging = get_logger(__name__)
+
+
+class SNS:
+    """Class for sending notifications using the Amazon SNS service."""
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self):
+        self.notification = SNSPushNotification()
+        self.publish = SNSPublishMessage()
+        self.sms = SNSSandboxSMS()
+        self.topic = SNSTopics()
+        self.subscribe = SNSTopicSubscribe()
+
+    def _send_message(self, client, kwargs):
+        """Sends a message using the given client and message parameters and
+        returns the response."""
+        response = client.publish(kwargs)
+        if "MessageId" in response:
+            logging.info("Notification sent successfully")
+            return response
+
+
+class SendSMS(SNS):
```

Comment:
pass it as parameter, for achieving loosely coupled and attend in all places

try to make it as loosely coupled in all area's